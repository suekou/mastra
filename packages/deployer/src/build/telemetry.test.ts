import { describe, it, expect } from 'vitest';
import { fileURLToPath } from 'node:url';
import { dirname, join } from 'node:path';
import { getTelemetryBundler } from './telemetry';

describe('getTelemetryConfig', () => {
  const _dirname = dirname(fileURLToPath(import.meta.url));

  it.for([
    ['./plugins/__fixtures__/basic.js', true],
    ['./plugins/__fixtures__/basic-with-const.js', true],
    ['./plugins/__fixtures__/basic-with-import.js', true],
    ['./plugins/__fixtures__/basic-with-function.js', true],
    ['./plugins/__fixtures__/mastra-with-extra-code.js', true],
    ['./plugins/__fixtures__/empty-mastra.js', false],
    ['./__fixtures__/no-telemetry.js', false],
  ] as [string, boolean][])(
    'should be able to extract the telemetry config from %s',
    async ([fileName, hasCustomConfig]) => {
      const hasConfigResult = {
        hasCustomConfig: false,
      } as const;
      const bundle = await getTelemetryBundler(join(_dirname, fileName), hasConfigResult);

      const result = await bundle.generate({
        format: 'esm',
      });

      expect(result?.output[0].code).toMatchSnapshot();
      expect(hasConfigResult.hasCustomConfig).toBe(hasCustomConfig);
    },
  );
});
