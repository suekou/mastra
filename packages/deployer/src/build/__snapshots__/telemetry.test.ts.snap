// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./__fixtures__/no-telemetry.js 1`] = `
"const telemetry = {};

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/basic.js 1`] = `
"const telemetry = {
  enabled: true,
  serviceName: "my-app",
  export: {
    type: "otlp",
    endpoint: "http://localhost:4318"
    // SigNoz local endpoint
  }
};

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/basic-with-const.js 1`] = `
"const _telemetry = {
  enabled: true,
  serviceName: "my-app",
  export: {
    type: "otlp",
    endpoint: "http://localhost:4318"
    // SigNoz local endpoint
  }
};
const telemetry = _telemetry;

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/basic-with-function.js 1`] = `
"function getTelemetryConfig() {
  return {
    enabled: true,
    serviceName: "my-app",
    export: {
      type: "otlp",
      endpoint: "http://localhost:4318"
      // SigNoz local endpoint
    }
  };
}
const telemetry = getTelemetryConfig();

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/basic-with-import.js 1`] = `
"import { telemetryConfig } from '@/telemetry';

const telemetry = telemetryConfig;

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/empty-mastra.js 1`] = `
"const telemetry = {};

export { telemetry };
"
`;

exports[`getTelemetryConfig > should be able to extract the telemetry config from ./plugins/__fixtures__/mastra-with-extra-code.js 1`] = `
"const telemetry = {
  enabled: true
};

export { telemetry };
"
`;
